### PR TITLE
system-variables: add description for variable `tidb_opt_ordering_index_selectivity_threshold`

### PR DESCRIPTION
**Pull request by time-and-fate**
### What is changed, added or deleted? (Required)

Add description for the new system variable `tidb_opt_ordering_index_selectivity_threshold`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/issues/42060
https://github.com/pingcap/tidb/pull/42116

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by xuyifangreeneyes — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
 - 默认值：`1.0`
 - 表示传输 1 比特数据的网络净开销。该变量是[代价模型](/cost-model.md)内部使用的变量，**不建议**修改该变量的值。
 
+### `tidb_opt_ordering_index_selectivity_threshold` <span class="version-mark">从 v7.0 版本开始引入</span>
```
Comment (ran-huang) [inline system-variables.md:2600]:
```suggestion
### `tidb_opt_ordering_index_selectivity_threshold` <span class="version-mark">从 v7.0.0 版本开始引入</span>
```

File: system-variables.md [hunk 2]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
...
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：浮点数
+- 默认值：`0`
+- 范围：`[0, 1]`
+- 用于当 SQL 中存在 `ORDER BY` 和 `LIMIT` 子句且带有过滤条件时，控制优化器选择索引的行为。
+- 对于此类查询，优化器会考虑选择对应的索引来满足 `ORDER BY` 和 `LIMIT` 子句（即使这个索引并不满足任何过滤条件）。但是由于数据分布的复杂性，优化器可能在这种场景下选择不优的索引。
+- 该变量表示一个阈值。当存在索引能满足过滤条件，且其选择率估算值低于该阈值时，优化器会避免选择用于满足 `ORDER BY` 和 `LIMIT` 的索引，而优先选择用于满足过滤条件的索引。
+- 例如，当把该变量设为 `0` 时，优化器保持默认行为；当设为 `1` 时，优化器总是优先选择满足过滤条件的索引，避免选择满足 `ORDER BY` 和 `LIMIT` 的索引。
+- 在以下示例中，`t` 表共有 1000000 行数据。使用 `b` 列上的索引时，其估算行数是大约 8748 行，因此其选择率估算值大约是 0.0087。默认情况下，优化器选择了 `a` 列上的索引。而将该变量设为 `0.01` 之后，由于 `b` 列上的索引的选择率（0.0087）低于 0.01，优化器选择了 `b` 列上的索引。
```
Comment (ran-huang) [inline system-variables.md:2611]:
```suggestion
- 在以下示例中，`t` 表共有 1,000,000 行数据。使用 `b` 列上的索引时，其估算行数是大约 8,748 行，因此其选择率估算值大约是 0.0087。默认情况下，优化器选择了 `a` 列上的索引。而将该变量设为 `0.01` 之后，由于 `b` 列上的索引的选择率（0.0087）低于 0.01，优化器选择了 `b` 列上的索引。
```

File: system-variables.md [hunk 3]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
...
+
+```sql
+> explain select * from t where b <= 9000 order by a limit 1;
```
Comment (ran-huang) [inline system-variables.md:2614]:
```suggestion
> EXPLAIN SELECT * FROM t WHERE b <= 9000 ORDER BY a LIMIT 1;
```

File: system-variables.md [hunk 4]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
...
++-----------------------------------+---------+-----------+----------------------+--------------------+
+| Limit_12                          | 1.00    | root      |                      | offset:0, count:1  |
+| └─Projection_25                   | 1.00    | root      |                      | test.t.a, test.t.b |
+|   └─IndexLookUp_24                | 1.00    | root      |                      |                    |
+|     ├─IndexFullScan_21(Build)     | 114.30  | cop[tikv] | table:t, index:ia(a) | keep order:true    |
+|     └─Selection_23(Probe)         | 1.00    | cop[tikv] |                      | le(test.t.b, 9000) |
+|       └─TableRowIDScan_22         | 114.30  | cop[tikv] | table:t              | keep order:false   |
++-----------------------------------+---------+-----------+----------------------+--------------------+
+
+> set session tidb_opt_ordering_index_selectivity_threshold = 0.01;
```
Comment (ran-huang) [inline system-variables.md:2626]:
```suggestion
> SET SESSION tidb_opt_ordering_index_selectivity_threshold = 0.01;
```

File: system-variables.md [hunk 5]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
...
+
+> explain select * from t where b <= 9000 order by a limit 1;
```
Comment (ran-huang) [inline system-variables.md:2628]:
```suggestion
> EXPLAIN SELECT * FROM t WHERE b <= 9000 ORDER BY a LIMIT 1;
```

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 6]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
...
 
+### `tidb_opt_ordering_index_selectivity_threshold` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：浮点数
+- 默认值：`0`
+- 范围：`[0, 1]`
+- 用于当 SQL 中存在 `ORDER BY` 和 `LIMIT` 子句且带有过滤条件时，控制优化器选择索引的行为。
+- 对于此类查询，优化器会考虑选择对应的索引来满足 `ORDER BY` 和 `LIMIT` 子句（即使这个索引并不满足任何过滤条件）。但是由于数据分布的复杂性，优化器可能在这种场景下选择不优的索引。
```
Comment (Oreoxmt) [inline system-variables.md:2608]:
```suggestion
- 对于此类查询，优化器会考虑选择对应的索引来满足 `ORDER BY` 和 `LIMIT` 子句（即使这个索引并不满足任何过滤条件）。但是由于数据分布的复杂性，优化器在这种场景下可能会选择不优的索引。
```

File: system-variables.md [hunk 7]
```diff
@@ -2597,6 +2597,46 @@ mysql> desc select count(distinct a) from test.t;
...
+- 该变量表示一个阈值。当存在索引能满足过滤条件，且其选择率估算值低于该阈值时，优化器会避免选择用于满足 `ORDER BY` 和 `LIMIT` 的索引，而优先选择用于满足过滤条件的索引。
+- 例如，当把该变量设为 `0` 时，优化器保持默认行为；当设为 `1` 时，优化器总是优先选择满足过滤条件的索引，避免选择满足 `ORDER BY` 和 `LIMIT` 的索引。
+- 在以下示例中，`t` 表共有 1,000,000 行数据。使用 `b` 列上的索引时，其估算行数是大约 8,748 行，因此其选择率估算值大约是 0.0087。默认情况下，优化器选择了 `a` 列上的索引。而将该变量设为 `0.01` 之后，由于 `b` 列上的索引的选择率（0.0087）低于 0.01，优化器选择了 `b` 列上的索引。
```
Comment (Oreoxmt) [inline system-variables.md:2611]:
```suggestion
- 在以下示例中，`t` 表共有 1,000,000 行数据。使用 `b` 列上的索引时，其估算行数是大约 8,748 行，因此其选择率估算值大约是 0.0087。默认情况下，优化器选择了 `a` 列上的索引。而将该变量设为 `0.01` 之后，由于 `b` 列上的索引的选择率 (0.0087) 低于 0.01，优化器选择了 `b` 列上的索引。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**